dali/python/nvidia/dali/pipeline.py

```diff
@@ -680,6 +681,25 @@ def _init_pipeline_backend(self):
         self._backend_prepared = True
         self._names_and_devices = [(e.name, e.device) for e in self._graph_outputs]
 
+    def _check_ext_source_groups_in_graph(self):
+        def truncate_str(obj, max_len=103):
+            obj_str = str(obj)
+            if len(obj_str) <= max_len:
+                return obj_str
+            return obj_str[:max_len - 3] + "..."
+
+        graph_op_ids = set(op.id for op in self._ops)
+        for group in self._input_callbacks:
+            for i, op in enumerate(group.instances):
+                if op.id not in graph_op_ids:
+                    num_outputs = len(group.instances)
+                    source_str = truncate_str(group.source_desc.source)
+                    raise RuntimeError(
```

Comment:
When there is more than one ES it will be hard to guess which one causes this but I guess we cannot distinguish ES instances anyhow.

Reply:
I hope for `group.source_desc.source` to give some clue (depending how top-level object it is it may be of some value).